MovieViewer/FirebaseUserDataService.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Firebase.swift
+//  MovieViewer
+//
+//  Created by hvmark on 7/11/16.
+//  Copyright © 2016 Dang Quoc Huy. All rights reserved.
+//
+
+import Firebase
+
+//class FirebaseUserDataService: UserDataServiceProtocol {
+class FirebaseUserDataService {
+
+    static func userFavoriteMovie(username: String, movieID: Int) {
```

Comment:
This is very clever, Hieu. I think this is a fantastic way to make this work. I have two small comments.

1. How do you unfavorite a movie? This will only work for adding favorites; I'd like the ability to remove favorites after I've favorite something.
2. This is likely to be very slow, because you'll be waiting until Firebase comes back to give the user any sort of feedback. Another basic suggestion would be to structure this like the HTTP request APIs: give a success and a failure callback, instead of just an `complete` callback. That way you could be "optimistic": the UI could reflect the new favorite right away, and then fix it in the case of a failure callback.